apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/navigation-menu/edit.js

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { IconButton, ServerSideRender, Toolbar } from '@wordpress/components';
+import { BlockControls } from '@wordpress/editor';
+import { addQueryArgs } from '@wordpress/url';
+import { Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+
+const NavigationMenuEdit = ( { attributes } ) => {
+	const url = addQueryArgs( 'customize.php', {
+		'autofocus[panel]': 'nav_menus',
+		return: window.location.href,
+	} );
+
+	return (
+		<Fragment>
+			<BlockControls>
+				<Toolbar className="wp-block-a8c-navigation-menu-toolbar">
+					<IconButton
+						icon="edit"
+						label={ __( 'Edit Menu' ) }
+						href={ url }
+						className="components-toolbar__control"
+					/>
+				</Toolbar>
+			</BlockControls>
+			<ServerSideRender
```

Comment:
To prevent the user from navigating away by clicking on the menu, we can wrap the `ServerSideRender` component with a `Disabled` component which [disables all the pointer events](https://github.com/WordPress/gutenberg/blob/a57202e635ff0cd190a09128d27951620db9a870/packages/components/src/disabled/style.scss#L3).

This is what [Gutenberg does with the `video` block](https://github.com/WordPress/gutenberg/blob/a57202e635ff0cd190a09128d27951620db9a870/packages/block-library/src/video/edit.js#L284) to prevent the user from playing videos within the editor (something we replicated in the [VideoPress extension](https://github.com/Automattic/jetpack/blob/96a9b033147b723eb1e6f5052151dc01c4f3b165/extensions/blocks/videopress/edit.js#L144)).

Reply:
Careful to namespace that to the block-editor view only. Some of the css selectors here are pretty generic. We may need to double-check what styling is enqueued and add more targeted selectors for that case. (Depending on how this is added this may also be enqueued in the published post/page view. This makes more sense for brand new blocks.).

Reply:
Rather than wrap in another component, I've just added the pertinent CSS to our block styles, which works perfectly. I've also made sure to enforce the styles' scope in 5fa1502f39.